feat: add TargetLimit and EnforcedTargetLimit (#3571)
prometheus-operator/prometheus-operator: Documentation/api.md

| queryLogFile | QueryLogFile specifies the file to which PromQL queries are logged. Note that this location must be writable, and can be persisted using an attached volume. Alternatively, the location can be set to a stdout location such as `/dev/stdout` to log querie information to the default Prometheus log stream. This is only available in versions of Prometheus >= 2.16.0. For more details, see the Prometheus docs (https://prometheus.io/docs/guides/query-log/) | string | false |
| enforcedSampleLimit | EnforcedSampleLimit defines global limit on number of scraped samples that will be accepted. This overrides any SampleLimit set per ServiceMonitor or/and PodMonitor. It is meant to be used by admins to enforce the SampleLimit to keep overall number of samples/series under the desired limit. Note that if SampleLimit is lower that value will be taken instead. | *uint64 | false |

> | enforcedTargetLimit | EnforcedTargetLimit defines global limit on number of scraped targets. This overrides any TargetLimit set per ServiceMonitor or/and PodMonitor. It is meant to be used by admins to enforce the TargetLimit to keep overall number of targets under the desired limit. Note that if TargetLimit is lower that value will be taken instead. | *uint64 | false |

Reviewer: Shouldn't it say `Note that if TargetLimit is *higher* that value will be taken instead.`?

Author: Done

| selector | Selector to select Endpoints objects. | [metav1.LabelSelector](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#labelselector-v1-meta) | true |
| namespaceSelector | Selector to select which namespaces the Endpoints objects are discovered from. | [NamespaceSelector](#namespaceselector) | false |
| sampleLimit | SampleLimit defines per-scrape limit on number of scraped samples that will be accepted. | uint64 | false |

> | targetLimit | TargetLimit defines per-scrape limit on number of scraped targets that will be accepted. | uint64 | false |

Reviewer: ```suggestion
| targetLimit | TargetLimit defines a limit on the number of scraped targets that will be accepted. | uint64 | false |
```

Author: Done
